### PR TITLE
Restore-CohesityRemoteMSSQLObject-Wrapper.ps1 is newly to Restore the DB

### PR DESCRIPTION
**Pull request by venmuthu**
Restore-CohesityRemoteMSSQLObject-Wrapper.ps1 is newly to Restore the DB based on SqlHost, SqlObjectName and TargetHost.

Restore-CohesityRemoteMSSQLObject.ps1 : DbRestoreOverwritePolicy option is newly added

### CODE REVIEW

**Comment by shubhamkr-cohesity** [comment]
@pyashish Can you please review this?

File: samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1 [hunk 1]
```diff
@@ -0,0 +1,148 @@
+<#
+    Use the generic cmdlet to create an MSSQL remote restore task
+
+    Example usage:
+        Restore-CohesityRemoteMSSQLObject-Wrapper.ps1 -JobId 1234 -SqlHost x.x.x.x -SqlObjectName "MSSQLSERVER/database" -TargetHost y.y.y.y -CaptureTailLogs:$false -NewDatabaseName database_new -NewInstanceName SQL2016 -TargetDataFilesDirectory "C:\" -TargetLogFilesDirectory "C:\temp" -DbRestoreOverwritePolicy:$true
```
Comment (pyashish) [inline samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1:5]: Can you write multiple examples to show restore from current vs remote cluster and stuff ?

File: samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1 [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+#>
+[CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $True, ConfirmImpact = "High")]
```
Comment (pyashish) [inline samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1:7]: If we have entries such as JobId, show how can they invoke other cmdlets and get this piped in here, if we are auto discovering it , probably mention the behavior (latest snapshot vs non latest) etc.

File: samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1 [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+$continuePagination = $true
+$searchTotalCount = 0
+
+# Loop through the result to fetch the specified SQL host id and parent id
+foreach ($record in $sqlRecords) {
+    while ($continuePagination) {
+        $searchURL = '/irisservices/api/v1/searchvms?from=' + $searchIndex + '&environment=SQL&entityTypes=kSQL&showAll=false&onlyLatestVersion=true&jobIds=' + $JobId
+        $searchVMResult = Invoke-RestApi -Method Get -Uri $searchURL
+
+        if ($Global:CohesityAPIStatus.StatusCode -ne 200) {
```
Comment (pyashish) [inline samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1:51]: What if there are multiple matches ?

Comment (pyashish) [reply samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1:51]: @KavishreeShanmugam11  did you get a chance to address these comments ?

File: samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1 (hunk 2 above)
Comment (KavishreeShanmugam11) [reply samples/RestoreSQLDB/Restore-CohesityRemoteMSSQLObject-Wrapper.ps1:7]: Done.

**Pull request merged**